rust_src/src/character.rs

```diff
@@ -18,3 +18,29 @@ defun!("max-char",
        0,
        ptr::null(),
        "Return the character of the maximum code.");
+
+/* Nonzero iff X is a character.  */
+pub fn CHARACTERP(x: LispObject) -> bool {
+    lisp::NATNUMP(x) && lisp::XFASTINT(x) <= MAX_CHAR
+}
+
+fn characterp(object: LispObject, ignore: LispObject) -> LispObject {
+    if CHARACTERP(object) {
+        LispObject::constant_t()
+    } else {
+        LispObject::constant_nil()
+    }
+}
+
+defun!("characterp",
+       Fcharacterp(x, y),
+       Scharacterp,
+       characterp,
+       1,
+       2,
+       ptr::null(),
+       "Return non-nil if OBJECT is a character.
+In Emacs Lisp, characters are represented by character codes, which
+are non-negative integers.  The function `max-char' returns the
+maximum character code.
+usage: (characterp OBJECT)");
```

Comment:
This looks correct. The reason you don't see the docstring in Remacs is a known bug, see #169.